### PR TITLE
Add CreateServiceLinkedRole to Index permissions

### PR DESCRIPTION
**Pull request by nhusepha9936**
*Issue #, if available:*
- Index permissions missing CreateServiceLinkedRole

*Description of changes:*
- Added CreateServiceLinkedRole 

*Tests:*
- Successfully assumed this role to create index

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-resourceexplorer2-defaultviewassociation/resource-role.yaml [hunk 1]
```diff
@@ -34,6 +34,10 @@ Resources:
                 - "resource-explorer-2:DisassociateDefaultView"
                 - "resource-explorer-2:GetDefaultView"
                 Resource: "*"
+              - Effect: Allow
+                Action:
+                - "iam:CreateServiceLinkedRole"
+                Resource: !Sub "arn:${AWS::Partition}:iam::${AWS::AccountId}:role/aws-service-role/resource-explorer-2.amazonaws.com/AWSServiceRoleForResourceExplorer"
```
Comment (thejacobseely) [inline aws-resourceexplorer2-defaultviewassociation/resource-role.yaml:40]: Permissions to create SLR should only be needed in `CreateIndex`.

**Review by thejacobseely — CHANGES_REQUESTED** [review]

File: aws-resourceexplorer2-defaultviewassociation/resource-role.yaml (hunk 1 above)
Comment (nhusepha9936) [reply aws-resourceexplorer2-defaultviewassociation/resource-role.yaml]: Removed SLR from the DefaultView and View

**Review by thejacobseely — APPROVED** [review]
Looks good, thanks!

**Review by aayachit — COMMENTED** [review]
LGTM

**Pull request merged**